pkg/controllers/workgenerator/controller.go

```diff
@@ -779,20 +877,81 @@ func setBindingStatus(works map[string]*fleetv1beta1.Work, resourceBinding *flee
 			continue // ignore the deleting work
 		}
 
-		// Failed placements (resources that cannot be applied or failed to get available) will only appear when either
-		// the Applied or Available conditions (on the work object) are set to False
-		if appliedCond.Status != metav1.ConditionTrue || availableCond.Status != metav1.ConditionTrue {
+		// Populate the failed, diffed, and drifted placements based on the summarized status of the Applied,
+		// Available, and DiffReported conditions on all Work objects.
+		//
+		// Note (chenyu1): Fleet will only report apply/availability check failures, diffs, and drifts (as applicable)
+		// when all the Work objects have completed their apply ops, availability checks, and diff reporting, as dictated
+		// by the currently specified apply strategy (successful or not). This is to make sure that previously
+		// populated failures, diffs, and/or drifts will not leak into the current reportings.
+		switch {
+		case isReportDiffModeOn && diffReportedSummarizedStatus == workConditionSummarizedStatusTrue:
+			// The ReportDiff apply straregy is in use and all works have reported configuration
+			// differences.
+			//
+			// In this case, set diffed placements only; failed and drifted placements will not
+			// be set (apply/availability check failure and drifts cannot occur in report diff mode).
+			diffedManifests := extractDiffedResourcePlacementsFromWork(w)
+			diffedResourcePlacements = append(diffedResourcePlacements, diffedManifests...)
+		case isReportDiffModeOn:
+			// The ReportDiff apply strategy is in use but not all works have reported configuration
+			// differences.
+			//
+			// In this case, no diffed, failed, or drifted placements will be set (diff information present
+			// might be incomplete or stale; apply/availability check failure and drifts cannot occur in
+			// report diff mode).
+		case appliedSummarizedStatus == workConditionSummarizedStatusIncomplete:
+			// The ClientSideApply or ServerSideApply apply strategy is in use but some of the works have
+			// not been applied yet.
+			//
+			// In this case, no diffed, failed, or drifted placements will be set (as information present
+			// might be incomplete or stale).
+		case appliedSummarizedStatus == workConditionSummarizedStatusFalse:
+			// The ClientSideApply or ServerSideApply apply strategy is in use but some of the works have
+			// apply op failures.
+			//
+			// In this case, set failed, diffed, and drifted placements.
 			failedManifests := extractFailedResourcePlacementsFromWork(w)
 			failedResourcePlacements = append(failedResourcePlacements, failedManifests...)
-		}
-		// Diffed placements can only appear when the Applied condition is set to False.
-		if appliedCond.Status == metav1.ConditionFalse {
+
 			diffedManifests := extractDiffedResourcePlacementsFromWork(w)
 			diffedResourcePlacements = append(diffedResourcePlacements, diffedManifests...)
```

Comment:
why report diff when "appliedSummarizedStatus == workConditionSummarizedStatusFalse"?

Reply:
```
 Fleet will report such differences when:
	//
	// * The CRP uses the ReportDiff apply strategy, which instructs Fleet to compare the hub
	//   cluster manifests against the live resources without actually performing any apply op; or
	// * Fleet finds a pre-existing resource on the member cluster side that does not match its
	//   hub cluster counterpart, and the CRP has been configured to only take over a resource if
	//   no configuration differences are found.
```

Reply:
Hi Ryan! This is not for the report diff mode; apply op failures can also have diffs (when the takeover setting is set to `TakeOverIfNoDiff`.